mlflow/tracing/processor/base_mlflow.py

```diff
@@ -100,31 +105,6 @@ def on_end(self, span: OTelReadableSpan) -> None:
 
         super().on_end(span)
 
-    def _get_experiment_id_for_trace(self, span: OTelReadableSpan) -> str:
-        """
-        Determine the experiment ID to associate with the trace.
-
-        The experiment ID can be configured in multiple ways, in order of precedence:
-          1. An experiment ID specified via the span creation API i.e. MlflowClient().start_trace()
-          2. An experiment ID specified via `mlflow.tracing.set_destination`
-          3. An experiment ID of an active run.
-          4. The default experiment ID
-        """
-        from mlflow.tracing.provider import _MLFLOW_TRACE_USER_DESTINATION
-        from mlflow.tracking.fluent import _get_latest_active_run
-
-        if experiment_id := get_otel_attribute(span, SpanAttributeKey.EXPERIMENT_ID):
-            return experiment_id
-
-        if destination := _MLFLOW_TRACE_USER_DESTINATION.get():
-            if exp_id := getattr(destination, "experiment_id"):
-                return exp_id
-
-        if run := _get_latest_active_run():
-            return run.info.experiment_id
-
-        return _get_experiment_id()
```

Comment:
This is needed in OtelMetricsMixin and MlflowV3SpanProcessor; I've moved it out as a reusable utility method, since I think it'll be helpful in the future beyond these two classes